### DISCUSSION TITLE
Array type in hana

### DISCUSSION THREAD

**Discussion by lucas-garrido** [linq2db/linq2db#3301] [Q&A]
Hello ! 
Simple question , is array column type supported with hana provider ? 
i've try but got this error : Cannot convert value 'System.Byte[]' to type 'System.String[]'

**Comment by MaceWindu** [comment]
We don't have array tests for HANA so it could have issues, but usually it could be fixed by defining additional mappings.
If you provide failing code, probably I can help you with it.

**Comment by lucas-garrido** [comment]
the db field type is `VARCHAR(255) ARRAY`

my class field is `public string[] myField { get; set; }
`

**Comment by MaceWindu** [comment]
Ok, I have some news for you. Not sure you will like it, but it is possible to use arrays now with some tuning, see my message and code here https://github.com/linq2db/linq2db/pull/3303

I will convert it into built-in support in future releases.

**Reply by lucas-garrido** [reply]
sounds great ! no worries i will keep doing associations as usual until you release it
thanks for your work 🙂
